.github/workflows/CI-orchestrator.yml

```diff
@@ -30,10 +29,62 @@ jobs:
             echo "act=false" >> $GITHUB_OUTPUT
           fi
 
-  notify-slack:
+  set-overall-result:
     runs-on: ubuntu-latest
     needs: [build-test-job, test-coverage-job, lint-format-job, e2e-test-job, check-act]
     if: ${{ always() && needs.check-act.outputs.act == 'false' }}
+    outputs:
+      branch-name: ${{ steps.get-info.outputs.BRANCH_NAME }}
+      last-commit-sha: ${{ steps.get-info.outputs.LAST_COMMIT_SHA }}
+      pr-url: ${{ steps.get-info.outputs.PR_URL }}
+      overall-status: ${{ steps.set-status.outputs.OVERALL_STATUS }}
+    steps:
+      - name: Get target PR info
+        id: get-info
+        run: |
+          BRANCH_NAME=${{ github.head_ref || github.ref_name }}
+          echo BRANCH_NAME is ${BRANCH_NAME}
+          echo "BRANCH_NAME=${BRANCH_NAME}" >> $GITHUB_OUTPUT
+          ALL_PRS=$(curl -s --fail \
+                    -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+                    "https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls?state=open")         
+          LAST_COMMIT_SHA=$(echo ${ALL_PRS} | jq ".[] | select(.head.label == \"${GITHUB_REPOSITORY_OWNER}:${BRANCH_NAME}\")" | jq -r ".head.sha")
+          echo LAST_COMMIT_SHA is ${LAST_COMMIT_SHA}
+          echo "LAST_COMMIT_SHA=${LAST_COMMIT_SHA}" >> $GITHUB_OUTPUT
+          echo "LAST_COMMIT_SHA=${LAST_COMMIT_SHA}" >> $GITHUB_ENV
+          PR_URL=$(echo ${ALL_PRS} | jq ".[] | select(.head.label == \"${GITHUB_REPOSITORY_OWNER}:${BRANCH_NAME}\")" | jq -r ".html_url")
+          echo PR_URL is ${PR_URL}
+          echo "PR_URL=${PR_URL}" >> $GITHUB_OUTPUT
+      - name: Set overall result
+        id: set-status
+        run: |
+          if [ "${{ github.event.workflow_run.conclusion }}" == "cancelled" ]; then
+            OVERALL_STATUS="cancelled"
+          else
+            OVERALL_STATUS="success"
+            if [ "${{ needs.build-test-job.result }}" != "success" ] || 
+                [ "${{ needs.test-coverage-job.result }}" != "success" ] ||
+                [ "${{ needs.lint-format-job.result }}" != "success" ] || 
+                [ "${{ needs.e2e-test-job.result }}" != "success" ]; then
+              OVERALL_STATUS="failure"
+            fi
+          fi
+  
+          echo Setting overall result
```

Comment:
From the [docs](https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28) I see `state` can be one of:
- `error`/`failure`
- `pending`
- `success`
so I would place the curl call in the non-`cancelled` conditional branch (before `fi` at line 72); in the `cancelled` case, the state would be simply not set at all.

Reply:
Done